### PR TITLE
refactor(connected-position-strategy): use top/left instead of transforms for positioning

### PR DESCRIPTION
**Pull request by crisbeto**
Switches the `ConnectedPositionStrategy` to use `top` and `left` for positioning, instead of `translateX` and `translateY`. This avoids having to do extra work to prevent subpixel rendering issues, while freeing up the transforms to be used on animations or for offsets.

### CODE REVIEW

File: src/lib/core/overlay/position/connected-position-strategy.ts [hunk 1]
```diff
@@ -22,10 +21,10 @@ export class ConnectedPositionStrategy implements PositionStrategy {
   private _dir = 'ltr';
 
   /** The offset in pixels for the overlay connection point on the x-axis */
-  private _offsetX: number = 0;
+  private _x: number = 0;
```
Comment (jelbourn) [inline src/lib/core/overlay/position/connected-position-strategy.ts]: Why rename these? They are still offset values.

File: src/lib/core/overlay/position/connected-position-strategy.ts [hunk 2]
```diff
@@ -227,18 +226,11 @@ export class ConnectedPositionStrategy implements PositionStrategy {
...
   private _setElementPosition(element: HTMLElement, overlayPoint: Point) {
-    // Round the values to prevent blurry overlays due to subpixel rendering.
-    let x = Math.round(overlayPoint.x);
-    let y = Math.round(overlayPoint.y);
-
-    // TODO(jelbourn): we don't want to always overwrite the transform property here,
-    // because it will need to be used for animations.
-    applyCssTransform(element, `translateX(${x}px) translateY(${y}px)`);
+    element.style.left = overlayPoint.x + 'px';
+    element.style.top = overlayPoint.y + 'px';
```
Comment (jelbourn) [inline src/lib/core/overlay/position/connected-position-strategy.ts]: Should we not still round these, or do browsers simply ignore subpixel top/left?

File: src/lib/select/select.spec.ts [hunk 3]
```diff
@@ -527,6 +527,8 @@ describe('MdSelect', () => {
       fixture.detectChanges();
 
       const overlayPane = overlayContainerElement.children[0] as HTMLElement;
+      overlayPane.style.position = 'absolute';
```
Comment (jelbourn) [inline src/lib/select/select.spec.ts]: Why is this necessary?

**Comment by jelbourn** [comment]
@kara should also take a look

File: src/lib/select/select.spec.ts (hunk 3 above)
Comment (crisbeto) [inline src/lib/select/select.spec.ts]: It's because it doesn't look like we import all of the styles in the unit tests. This means that the overlay is `position: static`. It wasn't an issue with the transform approach since they work on all of the positions.

File: src/lib/core/overlay/position/connected-position-strategy.ts (hunk 1 above)
Comment (crisbeto) [inline src/lib/core/overlay/position/connected-position-strategy.ts]: I guess they still looked pretty similar to the transforms. I'll put them back.

File: src/lib/select/select.spec.ts (hunk 3 above)
Comment (jelbourn) [inline src/lib/select/select.spec.ts]: Can you just add a comment mentioning that?

**Comment by crisbeto** [comment]
Rebased and addressed the feedback.

**Comment by jelbourn** [comment]
@kara will this conflict with your select alignment PR?

**Comment by crisbeto** [comment]
Seems like a bunch of tests are failing on this one now. I'll take a better look into it tomorrow.

**Comment by crisbeto** [comment]
Tests are sorted out now.
